src/System.CommandLine.Tests/Binding/ModelBinderTests.cs

```diff
@@ -265,6 +265,23 @@ public void Parse_result_can_be_used_to_modify_an_existing_instance_without_doin
             instance.IntOption.Should().Be(123);
         }
 
+        [Fact]
+        public void Modify_an_existing_instance_should_keep_all_default_values_if_no_argument_matches_option()
+        {
+            var parser = new Parser(new Command("the-command"));
+
+            var instance = new ClassWithComplexTypes();
+            var bindingContext = new BindingContext(parser.Parse("the-command"));
+            var binder = new ModelBinder(typeof(ClassWithComplexTypes));
+
+            binder.UpdateInstance(instance, bindingContext);
+
+            instance.BoolOption.Should().Be(default);
+            instance.IntOption.Should().Be(default);
+            instance.ListOptionDefaultNull.Should().BeNull();
+            instance.ListOptionDefaultEmpty.Should().BeEmpty();
```

Comment:
You can simplify the assertions and make the intent a little clearer by doing this:

```cs
instance.Should().BeEquivalentTo(new ClassWithComplexTypes());
```